### PR TITLE
Use perlcritic wrapper and plugins from common repo

### PR DESCRIPTION
**Pull request by perlpunk**
Also create a github workflow for it.

Issue: https://progress.opensuse.org/issues/155188

### CODE REVIEW

File: .github/workflows/perl-critic.yml [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+
+jobs:
+  perl-critic-checks:
+    runs-on: ubuntu-latest
+    name: "Perlcritic"
+    container:
+      image: perldocker/perl-tester
+    steps:
+      - uses: actions/checkout@v4
+      - run: make test-critic
```
Comment (kalikiana) [inline .github/workflows/perl-critic.yml:14]:
```
Policy "Perl::Critic::Policy::HashKeyQuotes" is not installed.
Policy "Perl::Critic::Policy::RedundantStrictWarning" is not installed.
```

I see those in CI 🤔

Comment (perlpunk) [reply .github/workflows/perl-critic.yml:14]: Thanks, I missed that. Makefile needs to run tools/perlcritic instead. pushed. Now we should see failures and I'm currently trying to fix them

**Comment by perlpunk** [comment]
Ok, while trying to fix more perlcritic complaints I found out that the plugins actually need more fixes.
I wish those plugins had been tested on our code base before.

**Comment by perlpunk** [comment]
After
* https://github.com/os-autoinst/os-autoinst-common/pull/54
I updated the subrepo.  I added some exceptions now for RedundantStrictWarning, I think that's good enough.

**Comment by perlpunk** [comment]
https://app.circleci.com/pipelines/github/perlpunk/openQA/357/workflows/d8ac13a9-ce17-413a-bbc2-8b9779b9e1a2/jobs/2482
```
[13:16:03] t/05-scheduler-full.t .. 1/? 
    #   Failed test 'jobs dispatched to different workers'
    #   at t/05-scheduler-full.t line 119.
    #          got: undef
    #     expected: anything else

    #   Failed test 'each of the two jobs allocated to one of the workers'
    #   at t/05-scheduler-full.t line 120.
    #          got: undef
    #     expected: anything else
    # 0
    # Looks like you failed 2 tests of 4.
[13:16:03] t/05-scheduler-full.t .. 2/? 
#   Failed test 'Scheduler worker job allocation'
#   at t/05-scheduler-full.t line 128.
...
```
This seems unrelated. I'll retrigger

**Comment by perlpunk** [comment]
Created https://progress.opensuse.org/issues/165198 for the unrelated failing test

**Review by okurz — APPROVED** [review]

**Comment by perlpunk** [comment]
So there seems to be a problem with circleci - I retriggered the failing test and it passed now, but the links in the github checks still link to the older run.
Also the codecov upload failed with the typical token error.
Let's see what happens when I rebase to trigger a completely new run

**Pull request merged**